analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Pattern.cs

```diff
@@ -97,8 +98,9 @@ private static ProgramState ProcessDeclaration(SymbolicContext context, ISymbol
         private static BoolConstraint BoolContraintFromConstant(ProgramState state, IIsPatternOperationWrapper isPattern) =>
             state[isPattern.Value] is { } value
             && isPattern.Pattern.WrappedOperation.Kind == OperationKindEx.ConstantPattern
-            && ConstantCheck.ConstraintFromValue(IConstantPatternOperationWrapper.FromOperation(isPattern.Pattern.WrappedOperation).Value.ConstantValue.Value) is BoolConstraint boolPattern
-            && PatternBoolConstraint(value, boolPattern) is { } newConstraint
+            && ConstantCheck.ConstraintFromConstantValue(IConstantPatternOperationWrapper.FromOperation(isPattern.Pattern.WrappedOperation).Value.ToSonar()) is { } constantValue
+            && constantValue.HasConstraint<BoolConstraint>()
+            && PatternBoolConstraint(value, constantValue.Constraint<BoolConstraint>()) is { } newConstraint
```

Comment:
```suggestion
            && constantValue.Constraint<BoolConstraint>() is { } boolConstantConstraint
            && PatternBoolConstraint(value, boolConstantConstraint) is { } newConstraint
```

Reply:
I don't like the dependency to `ConstantCheck.ConstraintFromConstantValue`. We can also retrieve the Constraint from `IConstantPatternOperationWrapper.FromOperation(isPattern.Pattern.WrappedOperation).Value` directly.

Reply:
Can `ConstantCheck.ConstraintFromConstantValue` be made `private` now?